lib/gui/components/modal/services/modal.js

```diff
@@ -55,23 +59,33 @@ module.exports = function($uibModal, $q) {
     return {
       close: modal.close,
       result: $q((resolve, reject) => {
-        modal.result
-          .then(resolve)
-          .catch((error) => {
+        modal.result.then((value) => {
+          AnalyticsService.logEvent('Modal accepted', value);
+
+          resolve(value);
+        }).catch((error) => {
+
+          // For some annoying reason, UI Bootstrap Modal rejects
+          // the result reason if the user clicks on the backdrop
+          // (e.g: the area surrounding the modal).
+          if (error === 'escape key press' || error === 'backdrop click') {
+            AnalyticsService.logEvent('Modal rejected', {
+              method: error
+            });
 
             // Bootstrap doesn't 'resolve' these but cancels the dialog;
-            // therefore call 'resolve' here applied to 'false'.
-            if (error === 'escape key press' || error === 'backdrop click') {
-              resolve();
+            // therefore call 'resolve' here.
+            resolve();
+
+          } else {
+            AnalyticsService.logEvent('Modal rejected', {
+              value: error
```

Comment:
Here it's the value with which the modal was rejected (`reject(error)`). On the `method: error`, the modal is rejected with the value `undefined`, but I thought it would be useful to know that the rejection is caused by the user hitting escape or clicking outside the modal. I can add an explicit `value: undefined` there too.

On a different note, I do feel that the events I came up are very unstructured. I didn't really had any better ideas on how to structure them, if you want to think about them a little and help out with better ways of making them more consistent, I'm open to ideas @lurch & @jviotti